noether/src/segmentation_server.cpp

```diff
@@ -0,0 +1,169 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+#include <noether_msgs/SegmentAction.h>
+#include <mesh_segmenter/mesh_segmenter.h>
+
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/io/vtk_lib_io.h>
+
+#include <vtkPolyDataNormals.h>
+#include <vtkCleanPolyData.h>
+#include <vtkWindowedSincPolyDataFilter.h>
+#include <vtkSmoothPolyDataFilter.h>
+
+//#include <noether/noether.h>
+#include <vtk_viewer/vtk_utils.h>
+
+class SegmentationAction
+{
+protected:
+  ros::NodeHandle nh_;
+  actionlib::SimpleActionServer<noether_msgs::SegmentAction> server_;
+  std::string action_name_;
+  noether_msgs::SegmentFeedback feedback_;
+  noether_msgs::SegmentResult result_;
+
+public:
+  SegmentationAction(ros::NodeHandle nh, std::string name)
+    : server_(nh_, name, boost::bind(&SegmentationAction::executeCB, this, _1), false), action_name_(name), nh_(nh)
+  {
+    server_.start();
+    ROS_INFO("Segmentation action server online");
+  }
+
+  ~SegmentationAction(void) {}
+
+  void executeCB(const noether_msgs::SegmentGoalConstPtr& goal)
+  {
+    // Step 1: Load parameters
+    // TODO: Pass these in along with action
+    std::string file;
+    double curvature_threshold;
+    int min_cluster_size, max_cluster_size;
+    bool show_individually, save_outputs;
+    nh_.param<std::string>("/mesh_segmenter_client_node/filename", file, "");
+    nh_.param<int>("/mesh_segmenter_client_node/min_cluster_size", min_cluster_size, 500);
```

Comment:
By using the / in front of the parameter name you inherently only access the global version with this exact name. Why are you doing this? You could use a private node handle and then these parameters could be set for only this node. There might be multiple of these nodes running, each with different parameters.

Reply:
That's a good point. I'll fix that.